Add EcFlow Engine Feature to Workflow (#4025)
NOAA-EMC/global-workflow: docs/source/setup.rst

Reviewer: There are a few other locations in this RST where `setup_xml` should be changed to `setup_workflow`.  It's also mentioned in `configure.rst`, `errors_faq.rst`, and `gcafs.rst`.

::

>    ./setup_workflow.py $EXPDIR/$PSLOT

Reviewer 2: ```suggestion
   ./setup_workflow.py $EXPDIR/$PSLOT rocoto|ecflow
```
The signature of `setup_workflow.py` has changed and now requires a second positional argument of the `workflow_engine`.  Options are `rocoto|ecflow`.  Only rocoto is supported.
This information should be added in the documentation.

Example:

::

>    ./setup_workflow.py /some_safe_disk_area/Joe.Schmo/expdir/test

Reviewer 2: ```suggestion
   ./setup_workflow.py /some_safe_disk_area/Joe.Schmo/expdir/test rocoto
```

> Additional options for setting up Rocoto are available with `setup_workflow.py -h` that allow users to change the number of failed tries, number of concurrent cycles and tasks as well as Rocoto's verbosity levels.

Reviewer 2: ```suggestion
Additional options for setting up Rocoto or ecFlow are available with `setup_workflow.py -h` that allow users to change the number of failed tries, number of concurrent cycles and tasks as well as verbosity levels.

Presently, only the Rocoto workflow engine is supported.  EcFlow capabilities are a work in progress.
```
